eventcontent.go

```diff
@@ -519,6 +542,10 @@ func (v *levelJSONValue) assignIfExists(to *int64) {
 
 // Check if the user ID is a valid user ID.
 func isValidUserID(userID string) bool {
-	// TODO: Do we want to add anymore checks beyond checking the sigil and that it has a domain part?
-	return userID[0] == '@' && strings.IndexByte(userID, ':') != -1
+	return userIDRegexp.MatchString(userID)
```

Comment:
`isValidUserID` seems to be used in [power level event auth](https://github.com/matrix-org/gomatrixserverlib/blob/5ff38e2c2839cc21cc37d9e5fbaf85386ffdf5f4/eventauth.go#L512-L518), which means this is not safe (at least not before MSC2828). Right now it's possible to make user and room IDs with arbitrary unicode (spaces, emojis, everything). For example, `@🐈️:maunium.net` can be found in #dendrite-dev:matrix.org

I think filters aren't used for anything critical, but obviously validating them means you won't be able to filter user/room IDs containing non-ascii characters.